ui/app/src/modules/Content/Publish/PublishDialog.tsx

```diff
@@ -82,65 +89,94 @@ export const selectAllDeps = (setChecked: Function, items: Item[]) => {
+const submitMap = {
+  'admin': goLive,
+  'author': submitToGoLive
+};
+
+function PublishDialog(props: PublishDialogProps) {
+  const {
+    items,
+    scheduling = 'now',
+    onClose
+  } = props;
 
   const [open, setOpen] = React.useState(true);
-  const [dialog, setDialog] = useReducer((a, b) => ({ ...a, ...b }), dialogInitialState);
+  const [dialog, setDialog] = useReducer((a, b) => ({ ...a, ...b }), {
+    ...dialogInitialState,
+    'scheduling': scheduling
+  });
   const [publishingChannels, setPublishingChannels] = useState(null);
   const [publishingChannelsStatus, setPublishingChannelsStatus] = useState('Loading');
   const [checkedItems, setCheckedItems] = useState<any>(checkState(items));   // selected deps
   const [checkedSoftDep, _setCheckedSoftDep] = useState<any>({}); // selected soft deps
   const [deps, setDeps] = useState<DependenciesResultObject>();
   const [showDepsButton, setShowDepsButton] = useState(true);
   const [submitDisabled, setSubmitDisabled] = useState(false);
+  const [showDepsDisabled, setShowDepsDisabled] = useState(false);
   const [apiState, setApiState] = useState({
-    publishing: false,
     error: false,
     global: false,
     errorResponse: null
   });
 
-  const { formatMessage } = useIntl();
-
   const user = useSelector<GlobalState, GlobalState['user']>(state => state.user);
   const siteId = useSelector<GlobalState, GlobalState['sites']>(state => state.sites).active;
+  const userSitesRoles: String[] = user.rolesBySite[siteId];    // TODO: check why is not working with editorial-ice
+  const userRole = userSitesRoles.includes('admin') ? 'admin' : 'author';
+  const submit = submitMap[userRole];
+
+  const { formatMessage } = useIntl();
 
   useEffect(getPublishingChannels, []);
-  useEffect(setRef, [checkedItems, checkedSoftDep]);
+  useEffect(() => {
+    function setRef() {
+      const result = (
+        Object.entries({ ...checkedItems, ...checkedSoftDep })
+        .filter(([key, value]) => value === true)
+        .map(([key]) => key)
+      );
+      setSelectedItems(result);
+    }
+
+    setRef();
+  }, [checkedItems, checkedSoftDep]);
 
   function getPublishingChannels() {
     setPublishingChannelsStatus('Loading');
     setSubmitDisabled(true);
     fetchPublishingChannels(siteId)
-      .subscribe(
-        ({ response }) => {
-          setPublishingChannels(response.availablePublishChannels);
-          setPublishingChannelsStatus('Success');
-          setSubmitDisabled(false);
-        },
-        ({ response }) => {
-          setPublishingChannelsStatus('Error');
-          setSubmitDisabled(true);
-        }
-      );
+    .subscribe(
+      ({ response }) => {
+        setPublishingChannels(response.availablePublishChannels);
+        setPublishingChannelsStatus('Success');
+        setSubmitDisabled(false);
+      },
+      ({ response }) => {
+        setPublishingChannelsStatus('Error');
+        setSubmitDisabled(true);
+      }
+    );
```

Comment:
Move into the useEffect(`=> HERE <=`, [...])

Reply:
@rart the detail in here is that `getPublishingChannels` is passed down to PublishForm, if getPublishingChannels fail it shows a `retry` button, that calls the function again. I can change it to something like `onGetPublishingChannelsFail` and call the function in the PublishDialog component, but I'd still need to have the function out from the useEffect. Please let me know what you think